gateway/src/apicast/policy/conditional/operation.lua

```diff
@@ -0,0 +1,53 @@
+local setmetatable = setmetatable
+local assert = assert
+
+local TemplateString = require 'apicast.template_string'
+
+local default_value_type = 'plain'
+
+local _M = {}
+
+local mt = { __index = _M }
+
+local supported_ops = { ["=="] = 1 , ["!="] = 1 }
+
+local function is_supported(op)
+  return supported_ops[op]
+end
+
+local function create_template(value, type)
+  return TemplateString.new(value, type or default_value_type)
+end
+
+--- Initialize an operation
+-- @tparam string left Left operand
+-- @tparam[opt] string left_type How to evaluate the left operand ('plain' or
+--   'liquid'). Default: 'plain'.
+-- @tparam string op Operation
+-- @tparam[opt] string right Right operand
+-- @tparam string right_type How to evaluate the right operand ('plain' or
+--   'liquid'). Default: 'plain'.
+function _M.new(left, left_type, op, right, right_type)
+  assert(is_supported(op), 'Unsupported operation')
+
+  local self = setmetatable({}, mt)
+
+  self.op = op
+  self.templated_left = create_template(left, left_type)
+  self.templated_right = create_template(right, right_type)
+
+  return self
+end
+
+function _M:evaluate(context)
+  local left_value = self.templated_left:render(context)
+  local right_value = self.templated_right:render(context)
+
+  if self.op == '==' then
```

Comment:
I think we can have a dispatch table for operations like:


```lua

local operations = {
  ['=='] = function(left, right) return left == right end
}
```

And assign the comparator function during initialization for better JIT.

Reply:
Fixed the same way as the one above :+1: